### PR TITLE
feat(playground): Add Syntax Hightlighting and folding for RomeAst

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Summary
1. Resolved https://github.com/rome/tools/issues/2896
2. Add Syntax Highlighting and folding for RomeAst

## Test Plan
1. Run the playground locally, and check if everything else is alright.

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
https://user-images.githubusercontent.com/17974631/180465297-4dc72f08-a45f-4452-a4e4-d3bf4dd42b27.mp4

**Comment by IWANABETHATGUY** [comment]
grammar definition see https://github.com/IWANABETHATGUY/lang-rome-ast

**Review by xunilrj — APPROVED** [review]

**Comment by IWANABETHATGUY** [comment]
# Bundle Size
## Main
```
dist/assets/favicon.5e112f83.svg               1.10 KiB
dist/assets/rome_playground_bg.e1d31b08.wasm   2223.35 KiB
dist/assets/romeWorker.276a0a9f.js             6.61 KiB
dist/assets/prettierWorker.811eafcc.js         753.31 KiB
dist/assets/success.4a59e56f.svg               0.82 KiB
dist/assets/failed.13568d2b.svg                0.92 KiB
dist/assets/copy.184a94f2.svg                  1.08 KiB
dist/index.html                                0.48 KiB
dist/assets/index.074b2a9c.css                 13.77 KiB / gzip: 3.78 KiB
dist/assets/index.699ca3cd.js                  1320.72 KiB / gzip: 403.06 KiB
```

## Current 
```
dist/assets/favicon.5e112f83.svg               1.10 KiB
dist/assets/rome_playground_bg.e1d31b08.wasm   2223.35 KiB
dist/assets/romeWorker.276a0a9f.js             6.61 KiB
dist/assets/prettierWorker.811eafcc.js         753.31 KiB
dist/assets/success.4a59e56f.svg               0.82 KiB
dist/assets/failed.13568d2b.svg                0.92 KiB
dist/assets/copy.184a94f2.svg                  1.08 KiB
dist/index.html                                0.48 KiB
dist/assets/index.074b2a9c.css                 13.77 KiB / gzip: 3.78 KiB
dist/assets/index.cfeded99.js                  1323.17 KiB / gzip: 404.21 KiB
```

+ 3KB before gzip, 1KB after gzip

**Review by MichaReiser — COMMENTED** [review]
This is awesome! 

Can you explain what errors you get when you move the source of [lang-rome-ast](https://github.com/IWANABETHATGUY/lang-rome-ast) into the playground rather than having its own npm package? 

The playground is using it the same way as any other module which is why I don't see much reason why it should cause issues. It may need a couple of additional steps to integrate it in our build chain though.

File: website/playground/package.json [hunk 1]
```diff
@@ -36,6 +40,13 @@
 		"vite": "2.9.5",
 		"vite-plugin-svgr": "^2.2.0"
 	},
+	"resolutions": {
+		"@codemirror/state": "^6.0.1",
+		"@codemirror/view": "^6.1.0"
+	},
+	"overrides": {
+    "@codemirror/state": "6.0.1"
+  },
```
[the remark this answers is not in the document]
Comment (IWANABETHATGUY) [reply website/playground/package.json]: removed it.

**Comment by IWANABETHATGUY** [comment]
> 

I have tried to move the `codemirror-lang-rome` package to our playground directory, but the code folding and highlight will not work after that, I am still not figured out why.

**Comment by IWANABETHATGUY** [comment]
I will transform the code to the playground directory once I found the solution. 
You could merge it now, and the version of `codemirror-lang-rome-ast` is locked in a specific version, we don't need to worry about the upstream issue.

**Comment by IWANABETHATGUY** [comment]
https://github.com/IWANABETHATGUY/lang-rome-ast, the repo is public feel free to review it.

**Comment by MichaReiser** [comment]
> > 
> 
> I have tried to move the `codemirror-lang-rome` package to our playground directory, but the code folding and highlight will not work after that, I am still not figured out why.

Do you get any errors if you do so? 

> You could merge it now, and the version of codemirror-lang-rome-ast is locked in a specific version, we don't need to worry about the upstream issue.

Sounds good to me.

**Comment by IWANABETHATGUY** [comment]
> Do you get any errors if you do so?
Let me open another issue to track it
> Since you're at it. Do you think it would be possible to implement something similar for Rome's formatting IR?
Let me try

**Comment by ematipico** [comment]
Maybe we can move the repository under Rome's org?

**Pull request merged**